spec/fdk-harvest-adm.yaml

```diff
@@ -0,0 +1,148 @@
+    url: "https://fellesdatakatalog.brreg.no"
+    email: fellesdatakatalog@brreg.no
+  license:
+    name: License of API
+    url: "http://data.norge.no/nlod/no/2.0"
+externalDocs:
+  description: Dokumentasjon på Informasjonsforvaltning GitHub
+  url: https://informasjonsforvaltning.github.io/felles-datakatalog/datasett-katalog/api/
+tags:
+  - name: dataSource
+    description: Data Source Service
+servers:
+  # - url: 'https://fellesdatakatalog.brreg.no/api'
+  - url: "localhost:8000/api"
+paths:
+  /datasources:
+    post:
+      tags:
+        - dataSource
+      summary: Create a new data source
+      operationId: createDataSource
+      responses:
+        "201":
+          description: Created
+          headers:
+            Location:
+              schema:
+                type: string
+              description: url to the new data source
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/DataSource"
+    get:
+      tags:
+        - dataSource
+      summary: Query for data sources.
+      description: Returns a collection of matching data sources
+      operationId: getDataSources
+      parameters:
+        - name: name
+          in: query
+          description: A query string to match a data source name
+          required: false
+          schema:
+            type: string
+        - name: publisher
+          in: query
+          description: If you want to filter by organizationId of the publisher
+          required: false
+          schema:
+            type: string
+        - name: dataSourceType
+          in: query
+          description: If you want to filter by the data source type
+          required: false
+          schema:
+            type: string
+            enum:
+              - SKOS-AP-NO
+              - DCAT-AP-NO
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/DataSource"
+  /datasources/{id}:
+    get:
+      tags:
+        - dataSource
+      description: Get a specific data source by id
+      operationId: getDataSourceById
+      parameters:
+        - name: id
+          in: path
+          description: id
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/DataSource"
+    put:
+      tags:
+        - dataSource
+      description: Update a specific dataSource by id
+      operationId: updateDataSource
+      parameters:
+        - name: id
+          in: path
+          description: id
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: OK
```

Comment:
hvorfor gir post empty body med location mens put gir body? det skulle funke samme måte...

Reply:
201 Created brukes ved POST. Den er ikkje relevant for PUT. Ref https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

Reply:
ja status 201 mener created, og det gjelder til bar POST, riktig. Men jeg spurte om innihold av response. 
det er flere ting man kan besteme Post response å være - tom med Location, tom med self-lenke som body, også det er mulig å få tilbake selv dokument (samme som get), eller bruke 301 i stedet av 201 (så klienter faktisk får body av get)

I put, man erstatter ressurs. Så det er mulig å ha akkurat samme body som post, bortsett fra at status er 204 (no content i stedet av 201

Jeg anbefaler å unifisere: POST gir 201 uten body og med Location, og PUT gir 204 uten body. Put trenger ikke location header (fordi det er samme som request)